### PR TITLE
Fix tests and compile issues

### PR DESCRIPTION
**Pull request by ggwpez**

### CODE REVIEW

File: polkadot/node/core/pvf/common/src/lib.rs [hunk 1]
```diff
@@ -25,9 +25,18 @@ pub mod worker;
...
+#[doc(hidden)]
+pub mod __private {
+	pub mod sp_tracing {
+		pub fn try_init_simple() {
+			#[cfg(feature = "test-utils")]
+			{
+				let _ = ::sp_tracing::try_init_simple();
+			}
+		}
+	}
```
Comment (bkchr) [inline polkadot/node/core/pvf/common/src/lib.rs:38]: When I see this. uhhh...
```suggestion
	#[cfg(feature = "test-utils")]
	pub use sp_tracing::try_init_simple;
```

File: polkadot/node/core/pvf/common/src/worker/mod.rs [hunk 2]
```diff
@@ -41,7 +41,7 @@ macro_rules! decl_worker_main {
 		}
 
 		fn main() {
-			$crate::sp_tracing::try_init_simple();
+			$crate::__private::sp_tracing::try_init_simple();
```
Comment (bkchr) [inline polkadot/node/core/pvf/common/src/worker/mod.rs:44]:
```suggestion
			$crate::__private::try_init_simple();
```

**Review by bkchr — APPROVED** [review]

File: polkadot/node/core/pvf/common/src/lib.rs (hunk 1 above)
Comment (ggwpez) [reply polkadot/node/core/pvf/common/src/lib.rs:38]: It is used in `decl_worker_main!` like this: `$crate::sp_tracing::try_init_simple();` which fails when the crate that uses the macro does not have the `test-utils` feature.  
I therefore added a stub in that case.

**Review by liamaharon — APPROVED** [review]

File: polkadot/node/core/pvf/common/src/lib.rs (hunk 1 above)
Comment (ggwpez) [reply polkadot/node/core/pvf/common/src/lib.rs:38]: You can see the compile error here @bkchr  https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3438147

Comment (bkchr) [reply polkadot/node/core/pvf/common/src/lib.rs:38]: This stuff is even more broken than expected :roll_eyes:

Comment (ggwpez) [reply polkadot/node/core/pvf/common/src/lib.rs:38]: You think we can merge it like this?

Comment (bkchr) [reply polkadot/node/core/pvf/common/src/lib.rs:38]: @ggwpez just remove the feature here entirely. And just expose it always. Is the macro feature gated at all?

**Review by juangirini — APPROVED** [review]

**Comment by bkchr** [comment]
@ggwpez please also add to the top `Cargo.toml`:
```
resolver = "2"
```

Under `[workspace]`

File: polkadot/node/core/pvf/common/src/lib.rs (hunk 1 above)
Comment (ggwpez) [reply polkadot/node/core/pvf/common/src/lib.rs]: okay removed the feature gate.

**Pull request merged**